File: medplum-java/medplum-model/src/main/java/com/fhir/GuidanceResponse.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableGuidanceResponse.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface GuidanceResponse {
  Optional<List<CodeableConcept>> reasonCode();

  Optional<Reference> encounter();

  Optional<List<Extension>> modifierExtension();

  Optional<GuidanceresponseStatus> status();

  Optional<Meta> meta();

  Optional<List<Reference>> evaluationMessage();

  Optional<Reference> performer();

  Optional<List<Identifier>> identifier();

  Optional<uri> implicitRules();

  Optional<CodeableConcept> moduleCodeableConcept();

  Optional<Narrative> text();

  Optional<Reference> outputParameters();

  String resourceType();

  Optional<List<Annotation>> note();

  Optional<id> id();

  Optional<code> language();

  Optional<dateTime> occurrenceDateTime();

  Optional<String> moduleCanonical();

  Optional<Reference> subject();

  Optional<String> moduleUri();

  Optional<List<Extension>> extension();

  Optional<Identifier> requestIdentifier();

  Optional<List<DataRequirement>> dataRequirement();

  Optional<List<ResourceList>> contained();

  Optional<List<Reference>> reasonReference();

  Optional<Reference> result();

  static ImmutableGuidanceResponse.ResourceTypeBuildStage builder() {
    return ImmutableGuidanceResponse.builder();
  }
}

File: medplum-typescript/src/model/fhir/GuidanceResponse.ts
import { Annotation } from "./Annotation";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { DataRequirement } from "./DataRequirement";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { GuidanceresponseStatus } from "./GuidanceresponseStatus";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface GuidanceResponse {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    requestIdentifier: Identifier | null | undefined;
    identifier: Identifier[] | null | undefined;
    moduleUri: string | null | undefined;
    moduleCanonical: string | null | undefined;
    moduleCodeableConcept: CodeableConcept | null | undefined;
    status: GuidanceresponseStatus | null | undefined;
    subject: Reference | null | undefined;
    encounter: Reference | null | undefined;
    occurrenceDateTime: dateTime | null | undefined;
    performer: Reference | null | undefined;
    reasonCode: CodeableConcept[] | null | undefined;
    reasonReference: Reference[] | null | undefined;
    note: Annotation[] | null | undefined;
    evaluationMessage: Reference[] | null | undefined;
    outputParameters: Reference | null | undefined;
    result: Reference | null | undefined;
    dataRequirement: DataRequirement[] | null | undefined;
}
